umbraco-cloud/set-up/legacy-umbraco-visual-studio-setup.md

```diff
@@ -0,0 +1,212 @@
+
+_So what goes where?_
+
+Anything that is used within Umbraco, like plugins and configuration, should by default be placed in the `*.Web` project. Here is a list of other elements that you want to place in the `*.Web` project:
+
+* Website assets like css, JavaScript and related images
+* Views, Partial Views and Partial View Macros
+* Configuration (web.config and all the Umbraco specific or related config files in ~/Config/)
+* Usercontrol ascx-files
+* Plugins (typically located in App_Plugins)
+* Meta data (the files that Umbraco Deploy uses in the folder ~/Data/Revision/)
+
+Media files will also be placed under the `*.Web` folder and you will be able to see these through Visual Studio, because a Website project shows all files on disk by default. Media files from the /Media/ folder should not be committed to the git repository, but more on that in the next section about 'What should be committed'.
+
+We recommend placing all your code in the `*.Core` project (instead of, for example, using App_Code for that). This includes, but is not limited to:
+
+* Controllers for MVC, Web Api
+* Controllers for Umbraco Plugins, Surface, API
+* Models and ViewModels
+* Data Access (the `*.Core` project references Umbraco so you can use the Umbraco datalayer as needed)
+* Extensions methods
+
+### Using ModelsBuilder and IntelliSense
+
+In order to use ModelsBuilder with IntelliSense in Visual Studio, you will need to make a couple of configuration changes to the web.config file of your `*.Web` project. This is to ensure that the models produced by ModelsBuilder are stored in the right place for compilation.
+1. Make sure ModelsBuilder.Enable is set to true (default): `<add key="Umbraco.ModelsBuilder.Enable" value="true" />`
+2. Set the Mode to `AppData` or `LiveAppData`. This will ensure you can use ModelsBuilder with Visual Studio. So in your Web.config, you should to have: `<add key="Umbraco.ModelsBuilder.ModelsMode" value="AppData" />`
+3. Create a directory called "Models" in your App_Code folder in the `*.Web` directory of your site. Then add: `<add key="Umbraco.ModelsBuilder.ModelsDirectory" value="~/App_Code/Models/" />` to Web.config.
+
+This will make the models of your Document Types available with IntelliSense in Visual Studio.
+[You can read more about configuring ModelsBuilder here.](/11/umbraco-cms/reference/templating/modelsbuilder/)
+
+{% hint style="info" %}
+
+If you are using the [Visual Studio Extension for ModelsBuilder](https://marketplace.visualstudio.com/items?itemName=ZpqrtBnk.UmbracoModelsBuilderExtension&ssr=false) and get the error message "Unauthorized" when generating models, you will need to use or create a backoffice user in your local installation, and supply the credentials for this user in the Visual Studio options. This is necessary because the extension is not able to authenticate against Umbraco Id.
```

Comment:
```suggestion
Are using the [Visual Studio Extension for ModelsBuilder](https://marketplace.visualstudio.com/items?itemName=ZpqrtBnk.UmbracoModelsBuilderExtension&ssr=false) and getting the error message Unauthorized when generating models? You'll need to use or create a backoffice user in your local installation. You then need to supply the credentials for this user in the Visual Studio options. This is necessary because the extension is not able to authenticate against Umbraco Id.
```